### ISSUE TITLE
Url_Key not regenerated for child categories

### ISSUE THREAD

**Issue by sebfie** [olegkoval/magento2-regenerate_url_rewrites#166]
**Describe the bug**
When running `bin/magento ok:urlrewrites:regenerate --store-id=6 --category-id=15 --no-reindex`

It does not regenerate child categories url_keys but just the parent category (in my case 15)

**Expected behavior**
It should regenerate url_key for all child categories

**To Reproduce**
Run `bin/magento ok:urlrewrites:regenerate --store-id=6 --category-id=15 --no-reindex`

**Additional context**
Here I do not understand why the level is 2 and not >=2 

https://github.com/olegkoval/magento2-regenerate_url_rewrites/blob/master/Model/RegenerateCategoryRewrites.php#LL259C7-L259C7

**Comment by pauldonnelly** [comment]
I am having the same issue. It seems only the top level categories are being generated but not the child categories.
I am using the latest version of the module and magento 2.4.6-p2

**Comment by Nuranto** [comment]
I agree. Also, I think there's a `->setStoreId($storeId)` missing in `_getCategoriesCollection`

**Comment by abdel-aouby** [comment]
the same for url_path it is generated only for the top-level categories or the current category when specify category id as option
it should as well generate it for all children

**Comment by olegkoval** [comment]
Fixed in 1.8.0 (same fix as #153): the category collection now includes the targeted category's descendants, not just the literal ID(s) given, so child categories' own url_key/url_path attributes get regenerated too - previously only their url_rewrite rows were refreshed via Magento's own generator cascade, never the url_key/url_path attribute values themselves.
